server/src/main/java/org/elasticsearch/health/node/ShardLimitsHealthIndicatorService.java

```diff
@@ -41,37 +42,35 @@
  */
 public class ShardLimitsHealthIndicatorService implements HealthIndicatorService {
 
+    static final String NAME = "shard_limits";
+
     private static final String UPGRADE_BLOCKED = "The cluster has too many used shards to be able to upgrade";
     private static final String UPGRADE_AT_RISK = "The cluster is running low on room to add new shards hence upgrade is at risk";
     private static final String HELP_GUIDE = "https://ela.st/max-shard-limit-reached";
-    static final String NAME = "shard_limits";
-    static final List<HealthIndicatorImpact> RED_INDICATOR_IMPACTS = List.of(
-        new HealthIndicatorImpact(NAME, "upgrade_blocked", 1, UPGRADE_BLOCKED, List.of(ImpactArea.DEPLOYMENT_MANAGEMENT))
-    );
-    static final List<HealthIndicatorImpact> YELLOW_INDICATOR_IMPACTS = List.of(
-        new HealthIndicatorImpact(NAME, "upgrade_at_risk", 2, UPGRADE_AT_RISK, List.of(ImpactArea.DEPLOYMENT_MANAGEMENT))
-    );
-
-    private static final BiFunction<String, String, Diagnosis> SHARD_LIMITS_REACHED_FN = (id, settingName) -> new Diagnosis(
+    private static final BiFunction<String, Setting<?>, Diagnosis> SHARD_LIMITS_REACHED_FN = (id, setting) -> new Diagnosis(
         new Diagnosis.Definition(
             NAME,
             id,
-            "The current value of the setting `"
-                + settingName
-                + "` is either about to be or already reached which will not allow adding more shards to the cluster.",
-            "Consider increasing the currently configured value or removing indices to clear up resources " + HELP_GUIDE,
+            "Elasticsearch is about to reach the maximum number of shards it can host.",
```

Comment:
The limit we are about to reach is not a real limit, rather it is based on the setting value. Let's try to find a way to reword this to more closely reflect that. Something like

"Elasticsearch is about to reach the maximum number of shards it can host, based on your current settings."